### DISCUSSION TITLE
Gulp task giving error "gulp failed with error: The process 'C:\npm\prefix\gulp.cmd' failed with exit code 1"

### DISCUSSION THREAD

**Discussion by sivakumar715** [gulpjs/gulp#2612] [Help]
we are running gulp task in the devops pipeline and it was working fine as of yesterday(august 2nd) but unexpectedly giving below error now.

![image](https://user-images.githubusercontent.com/54092242/128031470-b81c597e-405f-4982-b0ce-cae72c8be95a.png)

Node.js version - 11.15.0 

Gulp devops task version - 0

gulp task - https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/gulp?view=azure-devops

@brendan8c any suggestion?

please let us know if anyone else is aware of this issue.

**Comment by brend8c** [comment]
I think the problem is that you made a mistake while writing your code. For this reason, your gulp will not start.
Check your code. Pay special attention to the line from the screenshot.

![2021-08-04 15_05_43-128031470-b81c597e-405f-4982-b0ce-cae72c8be95a jpg ‎- Фотографии](https://user-images.githubusercontent.com/61526280/128178023-31d262d3-19e0-4c86-96f9-55191effe64a.png)

**Reply by sivakumar715** [reply]
thanks @brendan8c  for reply. actually this was working fine till august 2nd. The same code which was working before and its failing now at the gulp task with the above error.
I suspect the above parse error is from one of gulp dependencies but unable to find which package cause this issue.

**Note:** The Gulp task is working fine in local machine.
please suggest us if you are aware this issue.

**Comment by brend8c** [comment]
You can try deleting the node_modules folder.
Then check the cache.
`npm cache verify` Verifies the contents of the cache folder, garbage collecting any unneeded data, and verifying the integrity of the cache index and all cached data.
_`npm cache clean --force` delete the entire cache. I did not use it. But if you need to let this team be here._
And reinstall the dependencies `npm i`
This is just a guess what you can try to do.

**Reply by sivakumar715** [reply]
hi @brendan8c ,
i have deleted the "**node_modules**" from agent using delete files devops task(https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/delete-files?view=azure-devops) and installed npm again using npm devops task. 
But, still facing the same error.

even i ran the "npm cache verify" command under the source directory of node_modules. seems nothing is in cache. please suggest if anything is missed.
eg., _cd source directory 
npm cache verify_
![image](https://user-images.githubusercontent.com/54092242/128314105-05989c3d-62d2-459a-a4ef-a021f25fa696.png)

**Comment by brend8c** [comment]
I see that your task is aborted when Browserify starts to run.
Browserify starts to run something and comes up with an error.
It may be related to uglify-js.
You need to check the code. Perhaps up to this place
return `<$ {TEMPLATE_TAG}> $ {this.innerHTML} </ $ {TEMPLATE_TAG}>`; there is an error in the code. Line number 1652.

Perhaps The error came from a file written in ES6.
Can you replace uglify-js with a [terser](https://www.npmjs.com/package/terser)?
Try this. This should fix your problem.
Read [this](https://newbedev.com/uglification-failed-unexpected-character) additionally. The error is somehow related to ES6.

Or if you still want to use uglify you should try installing [gulp-uglify-es](https://www.npmjs.com/package/gulp-uglify-es).
[info 1](https://stackoverflow.com/questions/34398338/uglification-failed-unexpected-character)
[info 2](https://blog.csdn.net/tcr931117355/article/details/115295688)
